src/d/actor/d_a_obj_doguu.cpp

```diff
@@ -5,120 +5,811 @@
+    mLightInfluence.mColor.g = light_color[field_0x894][1];
+    mLightInfluence.mColor.b = light_color[field_0x894][2];
+    mLightInfluence.mPower = (s16)(field_0x8FC * 300.0f);
+    mLightInfluence.mFluctuation = 100.0f;
 }
 
 /* 00000168-00000188       .text CheckCreateHeap__FP10fopAc_ac_c */
-static BOOL CheckCreateHeap(fopAc_ac_c*) {
-    /* Nonmatching */
+static BOOL CheckCreateHeap(fopAc_ac_c* i_this) {
+    return static_cast<daObjDoguu_c*>(i_this)->CreateHeap();
 }
 
 /* 00000188-000007D0       .text CreateHeap__12daObjDoguu_cFv */
-void daObjDoguu_c::CreateHeap() {
-    /* Nonmatching */
+BOOL daObjDoguu_c::CreateHeap() {
+    J3DModelData* modelData = static_cast<J3DModelData*>(dComIfG_getObjectRes("Doguu", DOGUU_BDL_VGSMA));
+    JUT_ASSERT(0x160, modelData != NULL);
+    
+    J3DMaterialTable* bmt = static_cast<J3DMaterialTable*>(dComIfG_getObjectRes("Doguu", daObjDoguu_idx_table.bmt_vgsm[field_0x894]));
+    modelData->setMaterialTable(bmt, J3DMatCopyFlag_All);
+    field_0x6CC = mDoExt_J3DModel__create(modelData, 0x80000, 0x15220202);
+
+    if (field_0x6CC == NULL) {
+        return FALSE;
+    }
+
+    J3DAnmTevRegKey* brk_doguu = static_cast<J3DAnmTevRegKey *>(dComIfG_getObjectRes("Doguu", daObjDoguu_idx_table.brk[field_0x894]));
+    JUT_ASSERT(0x170, brk_doguu != NULL);
+    
+    if (mBrk.init(modelData, brk_doguu, true, J3DFrameCtrl::EMode_LOOP, 1.0f, 0, -1, false, 0) == 0) {
+      return FALSE;
+    }
+    
+    modelData = static_cast<J3DModelData*>(dComIfG_getObjectRes("Doguu", daObjDoguu_idx_table.bdl_vgsh[field_0x894]));
+    JUT_ASSERT(0x177, modelData != NULL);
+    field_0x6D0 =  mDoExt_J3DModel__create(modelData, 0,0x11020203);
+
+    if (field_0x6D0 == NULL) {
+        return FALSE;
+    }
+
+    J3DAnmTransform* bck_head = static_cast<J3DAnmTransform*>(dComIfG_getObjectRes("Doguu", daObjDoguu_idx_table.bck[field_0x894]));
+    JUT_ASSERT(0x17D, bck_head != NULL);
+    
+    if (mBck1.init(modelData, bck_head, true, J3DFrameCtrl::EMode_NONE, 1.0f, 0, -1, false) == 0) {
+        return FALSE;
+    }
+
+    J3DSkinDeform* deform = new J3DSkinDeform();
+
+    if (deform){
+        field_0x6D0->setSkinDeform(deform, 1);
+    } else {
+        return FALSE;
+    }
+    
+    modelData = static_cast<J3DModelData*>(dComIfG_getObjectRes("Doguu", DOGUU_BDL_VGSBA));
+    JUT_ASSERT(0x18f, modelData != NULL);
+    bmt =  static_cast<J3DMaterialTable *>(dComIfG_getObjectRes("Doguu", daObjDoguu_idx_table.bmt_vgsb[field_0x894]));
+    modelData->setMaterialTable(bmt, J3DMatCopyFlag_All);
+    field_0x6D4 = mDoExt_J3DModel__create(modelData, 0x80000,0x15220202);
+    
+    if (field_0x6D4 == NULL) {
+      return FALSE;
+    }
+    
+    JUTNameTab* nameTable = field_0x6D4->getModelData()->getJointName();
+    
+    for (u16 i = 0; i < field_0x6D4->getModelData()->getJointNum(); i++) {
+        const char* name = nameTable->getName(i);
+        if (strcmp("head1", name) == 0) {
+            mJointIdx = i;
```

Comment:
This could have the joint's name from the string in the member variable
```suggestion
            mHead1JntNo = i;
```